presto-accumulo/src/test/java/com/facebook/presto/accumulo/TestAccumuloDistributedQueries.java

```diff
@@ -0,0 +1,503 @@
+//                "SELECT 10");
+
+        // Tests for CREATE TABLE with UNION ALL: exercises PushTableWriteThroughUnion optimizer
+
+//        Function "UUID" not found
+//        this.assertCreateTableAsSelect(
+//                "test_union_all",
+//                "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders WHERE orderkey % 2 = 0 UNION ALL " +
+//                        "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders WHERE orderkey % 2 = 1",
+//                "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders",
+//                "SELECT count(*) FROM orders");
+
+//        Function "UUID" not found
+//        this.assertCreateTableAsSelect(
+//                getSession().withSystemProperty("redistribute_writes", "true"),
+//                "test_union_all",
+//                "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders UNION ALL " +
+//                        "SELECT UUID() AS uuid, DATE '2000-01-01', 1234567890, 1.23",
+//                "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders UNION ALL " +
+//                        "SELECT UUID() AS uuid, DATE '2000-01-01', 1234567890, 1.23",
+//                "SELECT count(*) + 1 FROM orders");
+
+//        Function "UUID" not found
+//        this.assertCreateTableAsSelect(
+//                getSession().withSystemProperty("redistribute_writes", "false"),
+//                "test_union_all",
+//                "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders UNION ALL " +
+//                        "SELECT UUID() AS uuid, DATE '2000-01-01', 1234567890, 1.23",
+//                "SELECT UUID() AS uuid, orderdate, orderkey, totalprice FROM orders UNION ALL " +
+//                        "SELECT UUID() AS uuid, DATE '2000-01-01', 1234567890, 1.23",
+//                "SELECT count(*) + 1 FROM orders");
+    }
+
+    @Override
+    public void testDelete()
+            throws Exception
+    {
+        try {
+            // TODO Deletes are not supported by the connector
+            super.testDelete();
+        }
+        catch (Exception e) {
+            assertUpdate("DROP TABLE test_delete");
+            assertEquals("This connector does not support updates or deletes", e.getMessage());
+        }
+    }
+
+    @Override
+    public void testInsert()
+            throws Exception
+    {
+        @Language("SQL") String query = "SELECT UUID() AS uuid, orderdate, orderkey FROM orders";
+
+        assertUpdate("CREATE TABLE test_insert AS " + query + " WITH NO DATA", 0);
+        assertQuery("SELECT count(*) FROM test_insert", "SELECT 0");
+
+        assertUpdate("INSERT INTO test_insert " + query, "SELECT count(*) FROM orders");
+
+        assertQuery("SELECT orderdate, orderkey FROM test_insert", "SELECT orderdate, orderkey FROM orders");
+        // Override because base class error: Cannot insert null row ID
```

Comment:
Nulls are represented in Accumulo as the absence of data.  Any non-row ID column can be null, it just isn't stored in Accumulo.  When the row is scanned, if there is no column family/qualifier entry in the row, the value of the cell is null.  The column chosen as the row ID always has to exist because there is no way to store the data in Accumulo in a way that it can be retrieved.

If you set `orderkey` to the row ID in this test case (and skip having a UUID column), then the below insert statement will fail because `orderkey` is null.

```
assertUpdate("INSERT INTO test_insert (uuid, orderdate) VALUES ('000001', DATE '2001-01-01')", 1);
```